File: src/main/org/nlogo/swing/InputDialog.java
package org.nlogo.swing;

public strictfp class InputDialog
    extends UserDialog {
  private final javax.swing.JTextField field = new javax.swing.JTextField();

  public InputDialog(java.awt.Frame owner, String title, String message, scala.Function1<String, String> i18n) {
    super(owner, title, i18n);
    addComponents(field, message);
  }

  public static String show(java.awt.Component owner, String title,
                            String message, String defaultInput, String[] options) {
    return (String) javax.swing.JOptionPane.showInputDialog
        (org.nlogo.awt.Utils.getFrame(owner),
            message, title,
            javax.swing.JOptionPane.QUESTION_MESSAGE,
            null, null, defaultInput);
  }

  public String showInputDialog() {
    java.awt.Rectangle r = getOwner().getBounds();
    setLocation
        (r.x + (r.width / 2) - (getWidth() / 2),
            r.y + (r.height / 2) - (getHeight() / 2));
    setVisible(true);
    if (selection == 0) {
      return field.getText();
    } else {
      return null;
    }
  }
}

File: src/main/org/nlogo/swing/InputDialog.scala
// (C) 2012 Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.swing

import java.awt.{ Component, Frame }
import org.nlogo.awt.Hierarchy.getFrame
import javax.swing.JOptionPane

object InputDialog {
  def show(owner: Component, title: String, message: String, defaultInput: String, options: Array[String]): String =
    javax.swing.JOptionPane.showInputDialog(
      getFrame(owner), message, title, JOptionPane.QUESTION_MESSAGE,
      null, null, defaultInput)
}

class InputDialog(owner: Frame, title: String, message: String, i18n: String => String)
extends UserDialog(owner, title, i18n) {
  private val field = new javax.swing.JTextField
  addComponents(field, message)
  def showInputDialog(): String = {
    val r = getOwner.getBounds
    setLocation(
      r.x + (r.width / 2) - (getWidth / 2),
      r.y + (r.height / 2) - (getHeight / 2))
    setVisible(true)
    if (selection == 0)
      field.getText
    else
      null
  }
}
